pkg/lib/declarative/indexconfig.go

```diff
@@ -0,0 +1,114 @@
+package declarative
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/operator-framework/operator-registry/pkg/containertools"
+	"github.com/operator-framework/operator-registry/pkg/image"
+	"github.com/operator-framework/operator-registry/pkg/image/containerdregistry"
+	"github.com/operator-framework/operator-registry/pkg/image/execregistry"
+	"github.com/operator-framework/operator-registry/pkg/lib/certs"
+	libregistry "github.com/operator-framework/operator-registry/pkg/lib/registry"
+	registry "github.com/operator-framework/operator-registry/pkg/registry"
+
+	"github.com/sirupsen/logrus"
+)
+
+// ConfigEditor is an implementation of IndexConfig
+type ConfigEditor struct {
+	Logger *logrus.Entry
+}
+
+// AddToConfig persists the bundle in it's package config
+func (c ConfigEditor) AddToConfig(request AddConfigRequest) error {
+
+	var reg image.Registry
+	var rerr error
+	switch request.ContainerTool {
+	case containertools.NoneTool:
+		rootCAs, err := certs.RootCAs(request.CaFile)
+		if err != nil {
+			return fmt.Errorf("failed to get RootCAs: %v", err)
+		}
+		reg, rerr = containerdregistry.NewRegistry(containerdregistry.SkipTLS(request.SkipTLS), containerdregistry.WithRootCAs(rootCAs))
+	case containertools.PodmanTool:
+		fallthrough
+	case containertools.DockerTool:
+		reg, rerr = execregistry.NewRegistry(request.ContainerTool, c.Logger, containertools.SkipTLS(request.SkipTLS))
+	}
```

Comment:
I can see why we might want to think about that. However, other interfaces used here also accept the parameters needed to construct the registry and construct it within themselves. And the constructions vary for example based on whether it's using the `buildTool` or the `pullTool` etc. I think it's totally reasonable to investigate the possibility of constructing the appropriate registries for the appropriate interface and inject them, but it sounds like the subject of a separate PR.

Reply:
> However, other interfaces used here also accept the parameters needed to construct the registry and construct it within themselves

I know there's precedent for this, but IMO not dependency injecting these makes the command harder to test; this is net new and I would like to avoid incurring the same issues from the start.

Reply:
@njhale is this what you had in mind https://github.com/operator-framework/operator-registry/pull/582/commits/5b18699efa188c9f408481aa5f56d0f0adfb2c4a
(I shall merge the commits if it is what we had in mind)

Reply:
yup! that's what I was looking for.